File: [71] Simplify Path/71.simplify-path.cpp
/*
 * @lc app=leetcode id=71 lang=cpp
 *
 * [71] Simplify Path
 */

// @lc code=start
class Solution {
public:
    string simplifyPath(string path) {
        vector<string> st;  // Use vector can reduce time for the last step to get res
        // stack<string> st;
        stringstream paths(path);
        string cur, res;
        while(getline(paths, cur, '/')) {
            if (cur == "" || cur == ".") continue;
            if (cur == "..") {
                if (!st.empty()) st.pop_back();
            } else st.push_back(cur);
        }
        if (st.empty()) return "/";
        for (auto tmp: st) {
            res += "/" + tmp;
        }
        return res;
    }
};
// @lc code=end

File: [71] Simplify Path/71.simplify-path.java
/*
 * @lc app=leetcode id=71 lang=java
 *
 * [71] Simplify Path
 */

// @lc code=start
class Solution {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack<>();
        for (String cur: path.split("/")){
            if (cur.equals("..")) {
                if (!stack.empty()) stack.pop();
            } else if (cur.length()>0 && !cur.equals(".")) stack.push(cur);
        }
        return "/"+String.join("/",stack);
    }
}
// @lc code=end

